File: llm/lib/tasks/extract-entity.mjs
// extract-entity — extract entities and typed relations from source text.
import { runJsonPrompt } from '../runner.mjs';

export async function run({ kbRoot, input }) {
  const sourcePath = input?.source_path || '';
  const body = input?.body || '';
  const { data } = await runJsonPrompt(kbRoot, 'extract-entity', { source_path: sourcePath, body });
  return {
    task: 'extract-entity',
    source_path: sourcePath,
    entities: Array.isArray(data.entities) ? data.entities : [],
    relations: Array.isArray(data.relations) ? data.relations : [],
  };
}

File: agent/weft_agent/tasks/extract_entity.py
"""extract-entity — extract entities and typed relations from source text
(port of llm/lib/tasks/extract-entity.mjs).
"""
from ..runner import run_json_prompt


def run(kb_root, input=None, output_path=None):
    input = input or {}
    source_path = input.get("source_path") or ""
    res = run_json_prompt(kb_root, "extract-entity", {
        "source_path": source_path,
        "body": input.get("body") or "",
    })
    data = res["data"]
    return {
        "task": "extract-entity",
        "source_path": source_path,
        "entities": data.get("entities") if isinstance(data.get("entities"), list) else [],
        "relations": data.get("relations") if isinstance(data.get("relations"), list) else [],
    }
